activedirectory.sgnl.yaml

```diff
@@ -0,0 +1,414 @@
+description: "LDAP as a System of Record"
+hostname: "{{Input Required: test.data.com}}"
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "LDAP-1.0.0"
+# Example Config:
+# {
+#   "baseDN":"dc=org,dc=example,dc=io",
+#   "certificateChain": ". . . base64 encoded certificate chain here . . "
+#   "entityConfig": {
+#     "User": {
+#       "query": "(&(objectCategory=user)(objectClass=user)(distinguishedName=*))"
+#     },
+#     "Computer": {
+#       "query": "(&(objectCategory=computer)(name=*))"
+#     },
+#     "Group": {
+#       "query": "(&(objectCategory=group)(objectClass=group)(distinguishedName=*))"
+#     },
+#     "GroupMember": {
+#       "memberOf": "Group",
+#       "query": "(&(memberOf={{CollectionId}})(objectCategory=user)(objectClass=user))",
+#       "memberUniqueIdAttribute": "memberObjectGUID",
+#       "memberOfUniqueIdAttribute": "groupObjectGUID"
+#     }
+#   },
+# }
+
+# Adapter Config Field Info:
+# [Required] baseDN (String): Specifies the Base Distinguished Name (DN) for the LDAP directory.
+
+# [Optional/Required] certificateChain (String): Contains the Base64 encoded certificate chain for secure communication with the LDAP server. (if LDAPS)
+
+# [Optional/Required] entityConfig (Object): Configurations for different types of entities within the LDAP directory.
+
+# For more information on LDAP queries: https://ldap.com/ldap-filters/
+# [Optional] entityConfig.User (Object): Configuration for user entities.
+# [Required] entityConfig.User.query (String): Specifies the LDAP query for retrieving user entities.
+
+# [Optional] entityConfig.Computer (Object): Configuration for computer entities.
+# [Required] entityConfig.Computer.query (String): Specifies the LDAP query for retrieving computer entities.
+
+# [Optional] entityConfig.Group (Object): Configuration for group entities.
+# [Required] entityConfig.Group.query (String): Specifies the LDAP query for retrieving group entities.
+
+# [Optional] entityConfig.GroupMember (Object): Configuration for group membership entities.
+# [Required] entityConfig.GroupMember.query (String): Specifies the LDAP query for retrieving group members.
+# [Required] entityConfig.GroupMember.memberOf (String): Specifies the group to which the members belong.
+# [Required] entityConfig.GroupMember.memberUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for members.
+# [Required] entityConfig.GroupMember.memberOfUniqueIdAttribute (String): Specifies the attribute used as a unique identifier for the group.
```

Comment:
This is fine for an `activedirectory` template. 

Please create an `ldap` template that contains detailed documentation on setting up `entityConfig` to ingest any kind of built-in/custom entity of various types (i.e. a standalone entity vs a member entity). Not a blocker for this PR.

Reply:
Added a task for `ldap` template: [#25545](https://app.shortcut.com/sgnl/story/25545/sor-improvements-for-active-directory-adapter)